source/core/parameters.cc

```diff
@@ -1388,12 +1407,38 @@ namespace Parameters
   {
     prm.enter_subsection("laser parameters");
     {
-      activate_laser       = prm.get_bool("enable");
+      activate_laser                = prm.get_bool("enable");
+      const std::string type_string = prm.get("type");
+      if (type_string == "exponential decay")
+        laser_type = LaserType::exponential_decay;
+      if (type_string == "material interface vof")
+        laser_type = LaserType::material_interface_vof;
       concentration_factor = prm.get_double("concentration factor");
       laser_power          = prm.get_double("power");
       laser_absorptivity   = prm.get_double("absorptivity");
       penetration_depth    = prm.get_double("penetration depth");
-      beam_radius          = prm.get_double("beam radius");
+
+      // Check if penetration depth is a strictly positive double and only
+      // defined when the laser is of exponential decay type
+      if (activate_laser)
+        {
+          if (laser_type == LaserType::exponential_decay)
+            {
+              AssertThrow(laser_type == LaserType::exponential_decay &&
+                            penetration_depth > 0.0,
+                          ParameterStrictlyGreaterThanError("penetration depth",
+                                                            penetration_depth,
+                                                            0.0));
+            }
+          else
+            {
+              AssertThrow(laser_type == LaserType::material_interface_vof &&
+                            penetration_depth == 0.0,
+                          PorousMediaDisabledError(penetration_depth));
+            }
```

Comment:
No need to do this assertion right, or are you actually using the penetration depth in the vof model? That's a bit confusing from my end

Reply:
This goes with my explanation on the comment of the exception.